portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,10 +44,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String content = request.getParameter("comment");
     String name = request.getParameter("name");
-    if (content.length() != 0) {
+    
+    if (content != null && name != null && content.length() != 0) {
       Comment comment = name.length() == 0 ? new Comment(content) : new Comment(name, content);
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("name", comment.getNAME());
+      commentEntity.setProperty("content", comment.getCONTENT());
+      commentEntity.setProperty("numLikes", comment.getNumLikes());
+      commentEntity.setProperty("timestamp", comment.getTIMESTAMP());
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Does this create a new DatastoreService instance at each call? If so, do we need to do that or is one datastore for this servlet instance fine?

Reply:
yep it looks like one instance for the servlet is sufficient